src/tests/unit/cpu/nodes/reorder_node_test.cpp

```diff
@@ -82,185 +211,373 @@ class ReorderCustomImplTestBase: public ::testing::Test {
+        auto outputNode = std::make_shared<MKLDNNPlugin::MKLDNNInputNode>(dstDims,
+                                                                          prec,
+                                                                          "Reorder_Output",
+                                                                          "Output",
+                                                                          cpuEngine,
+                                                                          weightsCache);
+
+        parentEdge = std::make_shared<MKLDNNPlugin::MKLDNNEdge>(inputNode, reorderNode, 0, 0);
+        childEdge = std::make_shared<MKLDNNPlugin::MKLDNNEdge>(reorderNode, outputNode, 0, 0);
+        parentEdge->changeStatus(MKLDNNPlugin::MKLDNNEdge::Status::NeedAllocation);
+        childEdge->changeStatus(MKLDNNPlugin::MKLDNNEdge::Status::NeedAllocation);
+        reorderNode->addEdge(parentEdge);
+        reorderNode->addEdge(childEdge);
+        auto rtParamsCache = std::make_shared<MKLDNNPlugin::MultiCache>(100);
+        inputDesc = srcBlockedDescCreator->createDesc(prec, srcDims);
+
+        const MKLDNNPlugin::CpuBlockedMemoryDesc outputDesc = dstBlockedDescCreator->createDesc(prec, dstDims);
+
+        auto parentMemory = std::make_shared<MKLDNNPlugin::MKLDNNMemory>(cpuEngine);
+        auto childMemory = std::make_shared<MKLDNNPlugin::MKLDNNMemory>(cpuEngine);
+        parentMemory->Create(inputDesc, nullptr);
+        childMemory->Create(outputDesc, nullptr);
+        parentEdge->reuse(parentMemory);
+        childEdge->reuse(childMemory);
+
+        reorderNode->setDescs(inputDesc, outputDesc);
+
+        std::vector<std::shared_ptr<MKLDNNPlugin::MKLDNNNode>> nodes{inputNode, reorderNode, outputNode};
+        for (auto& n : nodes) {
+            n->init();
+            n->getSupportedDescriptors();
+            n->initSupportedPrimitiveDescriptors();
         }
+        //Select inputDesc as primitive descriptor for reorder node
+        reorderNode->selectPrimitiveDescriptorByIndex(0);
+        stream = {cpuEngine};
+        return;
     }
-    std::vector<uint8_t> dstData;
-    std::vector<uint8_t> srcData;
+
+private:
+    std::shared_ptr<MKLDNNPlugin::MultiCache> rtParamsCache;
+    mkldnn::stream stream;
+    std::shared_ptr<MKLDNNPlugin::MKLDNNReorderNode> reorderNode;
+    std::shared_ptr<MKLDNNPlugin::MKLDNNNode> inputNode;
+    std::shared_ptr<MKLDNNPlugin::MKLDNNEdge> parentEdge;
+    std::shared_ptr<MKLDNNPlugin::MKLDNNEdge> childEdge;
+
+    MKLDNNPlugin::Shape srcDims;
+    MKLDNNPlugin::Shape dstDims;
+    LayoutType srcLayout;
+    LayoutType dstLayout;
+    InferenceEngine::Precision prec;
+    std::vector<std::vector<size_t>> inputShapes;
+    ngraph::PartialShape inputPartialShape;
+    BlockedDescCreator::CreatorsMap blockCreatorMap;
+    MKLDNNPlugin::CpuBlockedMemoryDesc inputDesc{InferenceEngine::Precision::FP32, MKLDNNPlugin::Shape{}};
```

Comment:
It looks like we do not need to store the most part of this variables in the test object, they can be just passed as parameters for the `buildReorderNode()` method.